desktop/libs/notebook/src/notebook/connectors/hiveserver2.py

```diff
@@ -434,6 +434,109 @@ def get_jobs(self, notebook, snippet, logs):
 
     return jobs
 
+  def _get_fetch_result_spk(self, _statement, snippet):
+    """HUE-8843"""
+    db = self._get_db(snippet, cluster=self.cluster)
+    statement = {
+      'statements_count': 1,
+      'statement': _statement,
+      'has_more_statements': False,
+    }
+    session = None
+    snippet['properties'] = {}
+    query = self._prepare_hql_query(snippet, statement['statement'], session)
+
+    try:
+      db.use(query.database)
+      handle = db.client.query(query, with_multiple_session=True)
+    except QueryServerException as ex:
+      raise QueryError(ex.message, handle=statement)
+
+    server_id, server_guid = handle.get()
+    spk_handle = {
+      'secret': server_id,
+      'guid': server_guid,
+      'operation_type': handle.operation_type,
+      'has_result_set': handle.has_result_set,
+      'modified_row_count': handle.modified_row_count,
+      'log_context': handle.log_context,
+      'session_guid': handle.session_guid
+    }
+    spk_handle.update(statement)
+    spk_snippet = {
+      "type": "sparksql",
+      'result': {
+        'handle': spk_handle,
+      }
+    }
+
+    import time
```

Comment:
Could we move it to the top of the file?

Reply:
Of course~